dotfiles/.config/git/hooks/pre-commit

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+# Pre-commit hook: validates branch, prevents protected commits, and runs pre-commit with auto-staging
+#
+# Features:
+# 1. Prevents commits to master/main in external repos without push access
+# 2. Warns if branch is not based on latest origin/master
+# 3. Runs pre-commit with auto-staging (re-stages files modified by formatters)
+
+set -e
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+# --- Part 1: Prevent master commits in external repos ---
+BRANCH=$(git branch --show-current 2>/dev/null)
+if [ "$BRANCH" = "master" ] || [ "$BRANCH" = "main" ]; then
+    REMOTE_URL=$(git remote get-url origin 2>/dev/null)
+
+    # Repos where master/main commits are forbidden (external repos without push access)
+    # Customize this list for your agent's commonly-used external repos
+    FORBIDDEN_PATTERNS=(
+        "gptme/gptme"
+        "gptme/gptme-contrib"
+        # Add other repos where you contribute but don't have push to master
+    )
```

Comment:
I think this should probably be reversed, have a list of repos where push to master is allowed (automatically including the agent repo itself) instead of forbidden.